### PR TITLE
remote datasets

### PR DESCRIPTION
**Pull request by suvarchal**
This PR adds remote datasets as 'batteries-included' to pyfesom2. 

This example notebook summarizes it:
[remote_datasets.ipynb](https://nbviewer.jupyter.org/github/FESOM/pyfesom2/blob/suvi_datasets/notebooks/remote_datasets.ipynb) 

immediate todos on review are to add a few doc strings.

feature wise i would like to add local caching and extend example notebook to show how to use it for CMIP6, but will leave this for later.

### CODE REVIEW

File: pyfesom2/datasets.py
Comment (koldunovn) [inline pyfesom2/datasets.py:81]: Please rename it to just `core`. Also in other places through the code :)

**Review by koldunovn — APPROVED** [review]
Looks good to me except the `lcore` naming, that is better to change to avoid an impressing we add some other dataset, that is different from the CORE :)

File: pyfesom2/datasets.py
Comment (suvarchal) [reply pyfesom2/datasets.py:81]: Okay, will also add atleast minimal doc strings for couple of non doc'd methods before merge.

**Comment by koldunovn** [comment]
Thanks, @suvarchal ! Should we wait until you add doc strings, or we are ready to merge?

**Comment by suvarchal** [comment]
> Thanks, @suvarchal ! Should we wait until you add doc strings, or we are ready to merge?

Few more mins please.

File: pyfesom2/datasets.py
Comment (suvarchal) [reply pyfesom2/datasets.py]: I am done, if you mark it resolved I will merge or you can :)

**Pull request merged**
